pyedflib/highlevel.py

```diff
@@ -756,6 +757,81 @@ def anonymize_edf(edf_file, new_file=None,
     return True
 
 
+def crop_edf(edf_file, *, new_file=None, new_start=None, new_stop=None,
+             verbose=True):
+    """Crop an EDF file to desired start/stop times.
+
+    Parameters
+    ----------
+    edf_file : str
+        The path to the EDF file.
+    new_file : str | None
+         The path to the new cropped file. If None (default), the input
+         filename appended with '_cropped' is used.
+    new_start : datetime.datetime
+        The new start timestamp. Can be None to keep the original start time.
+    new_stop : datetime.datetime
+        The new stop timestamp. Can be None to keep the original stop time.
+    verbose : bool
+        If True (default), print some details about the original and cropped file.
+    """
+    # Check input
+    assert isinstance(new_start, (datetime, type(None)))
+    assert isinstance(new_stop, (datetime, type(None)))
+
+    # Open the original EDF file
+    edf = pyedflib.EdfReader(edf_file)
+    signals_headers = edf.getSignalHeaders()
+    header = edf.getHeader()
+
+    # Define new start time
+    current_start = edf.getStartdatetime()
+    if new_start is None:
+        new_start = current_start
+    start_diff_seconds = (new_start - current_start).total_seconds()
+    assert current_start <= new_start
+
+    # Define new stop time
+    current_stop = current_start + timedelta(seconds=edf.getFileDuration())
+    current_duration = current_stop - current_start
+    if new_stop is None:
+        new_stop = current_stop
+    assert new_stop <= current_stop
+    stop_diff_from_start = (new_stop - current_start).total_seconds()
+
+    # Crop each signal
+    signals = []
+    for i in range(len(edf.getSignalHeaders())):
+        # Question: should we use `_get_sample_frequency` instead?
```

Comment:
Ping @skjerns what's the best way to get the sampling frequency?

Reply:
getSampleFrequency should be good for now